crates/biome_js_analyze/src/lint/nursery/use_trim_start_end.rs

```diff
@@ -0,0 +1,292 @@
+    /// - [String.prototype.trimStart()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimStart)
+    /// - [String.prototype.trimEnd()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd)
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = bar.trimLeft();
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = bar.trimRight();
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const foo = bar.trimStart();
+    /// ```
+    ///
+    /// ```js
+    /// const foo = bar.trimEnd();
+    /// ```
+    ///
+    pub UseTrimStartEnd {
+        version: "next",
+        name: "useTrimStartEnd",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintUnicorn("prefer-string-trim-start-end")],
+        fix_kind: FixKind::Safe,
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct UseTrimStartEndState {
+    member_name: String,
+    span: TextRange,
+    suggested_name: String,
+}
+
+impl Rule for UseTrimStartEnd {
+    type Query = Ast<JsCallExpression>;
+    type State = UseTrimStartEndState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let arguments = node.arguments().ok()?;
+        let args = arguments.args();
+
+        if !args.is_empty() {
+            // If arguments are present, it suggests this function call may not be intended for `String.trimStart()` or `String.trimEnd()`,
+            // as these methods do not accept parameters according to the specification:
+            // - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimStart#parameters
+            // - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd#parameters
+            return None;
+        }
+
+        let callee = node.callee().ok()?;
+        let (member_name, span, suggested_name) = match callee {
+            AnyJsExpression::JsComputedMemberExpression(callee) => {
+                let member = callee.member().ok()?;
+                let value = member.as_static_value()?;
+                let span = value.range();
+                let member_name = value.as_string_constant()?.to_string();
+                let suggested_name = generate_suggested_name(&member_name);
+                (member_name, span, suggested_name)
+            }
+            AnyJsExpression::JsStaticMemberExpression(callee) => {
+                let token = callee.member().ok()?.value_token().ok()?;
+                let span = token.text_range();
+                let member_name = token.text_trimmed().to_string();
+                let suggested_name = generate_suggested_name(&member_name);
+                (member_name, span, suggested_name)
```

Comment:
I was trying to understand **where** we check for `trimLeft` a `trimEnd`, and it turns out it is implicitly done at line 106, with `suggested_name?`. Here's the issue: you allocate a **new** string _before_ actually checking if `member_name` is actually `trimLeft` or `trimEnd`. 

This means that for each computed expression or static member expression, we allocate a string and then throw it away (for the majority of times). You should first check and then allocate the string.